### PR TITLE
Update llx_societe.sql

### PR DESCRIPTION
**Pull request by defrance**
add outstanding value

### CODE REVIEW

File: htdocs/comm/fiche.php [hunk 1]
```diff
@@ -128,6 +128,14 @@
 	if ($result < 0) setEventMessage($object->error,'errors');
 }
 
+// Update communication level
+if ($action == 'setOutstandingBill')
+{
+	$object->fetch($id);
+	$object->stcomm_id=GETPOST('OutstandingBill');
```
Comment (atm-maxime) [inline htdocs/comm/fiche.php]: Hi Charles.

I guess here should be the outstanding var, not stcomm_id :)
If I understand correctly this new field will allow user to input an outstanding amount per customer. Shouldn't this be just a calculated information displayed (sum of amount reamining to pay) ?

Regards

Comment (defrance) [inline htdocs/comm/fiche.php]: please see here the goal of the feature
http://www.patas-monkey.com/index.php/fr/actualites/54-encours-de-facturation
i have to add the rest of the code in 3.5 core

Comment (atm-maxime) [inline htdocs/comm/fiche.php]: Ok thanks ! That's a great evolution.
My point was just that I think there's a mistake in the var you are using before calling "set_OutstandingBill".

Thanks

**Comment by defrance** [comment]
Oups your right,
I'll take time this week-end to finish it
I allready have a version running for 3.4.0

**Comment by eldy** [comment]
Sorry i still did not understood goal.
Do you mean you want to show amount of other invoice not paid from same customer ?

**Comment by atm-maxime** [comment]
The goal is to be able to define an outstanding amount per customer. And in documents, this amount and the sum of all unpaid invoice will be displayed so the user can know if the customer has already exceeded his "authorized" amount. I had some customer questions about that. This will allow to the user to decide whether or not he accept a customer order knowing that the customer already owes money to the company.

**Comment by eldy** [comment]
Le 27/09/2013 17:24, Maxime Kohlhaas a écrit :

> The goal is to be able to define an outstanding amount per customer.
> And in documents, this amount and the sum of all unpaid invoice will
> be displayed so the user can know if the customer has already exceeded
> his "authorized" amount. I had some customer questions about that.
> This will allow to the user to decide whether or not he accept a
> customer order knowing that the customer already owes money to the
> company.
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/Dolibarr/dolibarr/pull/1240#issuecomment-25253182.

But what is difference between this amount and the "the sum of all
unpaid invoice"
The new amount is the limit allowed ?

## 

Eldy (Laurent Destailleur).

EMail: eldy@destailleur.fr
Web: http://www.destailleur.fr

Dolibarr (Project leader): http://www.dolibarr.org
To make a donation for Dolibarr project via Paypal: contact@destailleur.fr
AWStats (Author) : http://awstats.sourceforge.net
To make a donation for AWStats project via Paypal: contact@destailleur.fr
AWBot (Author) : http://awbot.sourceforge.net
CVSChangeLogBuilder (Author) : http://cvschangelogb.sourceforge.net

**Comment by defrance** [comment]
One is the virtual limit allowed 
The other is the sum of bills unpaid
It's informative values but critical...

Bien cordialement,
Charles-françois BENKE

**Comment by atm-maxime** [comment]
Yes this can be understood as a limit, but this can also be a simple indicator. I have worked with a company that buy to another a file refreshed weekly containing financial indicators about its customers. And among them, the estimated outstanding amount that allow to know the financial health of the customer.

**Comment by defrance** [comment]
OK
I have correct the bad value setting
and add the last modification to finish the feature

**Comment by defrance** [comment]
Travis have detect an error on htdocs/core/lib/functions.lib.php
"PEAR.Commenting.FunctionComment.SpacingBeforeTags"
but i have change nothing on this file

**Comment by eldy** [comment]
1) Ok, so i think we should use a name for new field that is more easier to understand.
I suggest "maxoutstanding"  (even if for some case some people may use it to store data coming from external source so not as a max)

2) For function into societe that calculate remain to pay on all invoices, please add this comment
/\* Accurate value of remain to pay is to sum remaintopay for each invoice 
                            $paiement = $invoice->getSommePaiement();
                            $creditnotes=$invoice->getSumCreditNotesUsed();
                            $deposits=$invoice->getSumDepositsUsed();
                            $alreadypayed=price2num($paiement + $creditnotes + $deposits,'MT');
                            $remaintopay=price2num($invoice->total_ttc - $paiement - $creditnotes - $deposits,'MT');
*/

3) Also 
$langs->trans("Currency".$conf->currency)
must not be used anymore
Instead, use the parameter of price function to add the $conf->currency (in same cell than amount)

**Comment by defrance** [comment]
1 / maybe "outstanding_limit" is better : a max can't be exceed, a limit can
2 /  My calculation does not take into account the alreadypayed value (they need a very too clomplex query)
3 / the currency display method used is the same that the price fields near in the file (ex : propal.php ligne 1969) , maybe need to change it for all too

I realise too that i hav'nt made the change on commande fiche.php, i have job this week-end!

**Comment by eldy** [comment]
1) Ok for ""outstanding_limit"

2) I know for 2, so just add the comment i suggest into code to warn and remind that you don't take this into account.

3) Yes, printing the currency after the price must be replaced everywhere with using the currency into price function. Reason is that in some country/currency like dollar, the symbol must be printed before. This is managed by the price function.

**Comment by defrance** [comment]
OK Changes made

**Comment by eldy** [comment]
Strange. There is a merge conflict but nothing was merged after your pr.

**Comment by defrance** [comment]
they are a Travis conflict at the top of the pull (error in function.lib.php who i haven't change)

**Comment by eldy** [comment]
Added manually

**Pull request closed without merging**